### PR TITLE
Adding Portfiles : FnaFNES

### PR DESCRIPTION
**Pull request by Damian2809**
# New Port for FNAF NES STYLED

## Game Information
- **Title**: FNAF NES STYLED
- **URL**:  https://tai5-5.itch.io/fnafnes

### CFW Tests
Ensure your game has been tested on all major CFWs:
- [X] AmberELEC
- [X] ArkOS
- [X] ROCKNIX
- [X] muOS

### Resolution Tests
Test all major resolutions:
- [ ] 480x320
- [X] 640x480
- [X] 720x720 (RGB30)
- [X] Higher resolutions (e.g., 1280x720)

## File Structure
- Your port should have the following structure:
  - portname/
    - port.json
    - README.md
    - screenshot.jpg
    - cover.jpg
    - gameinfo.xml
    - Port Name.sh
    - portname/
      - <portfiles here>

## Additional Resources
For an in-depth guide on creating a pull request, refer to: [PortMaster Game Packaging Guide](https://portmaster.games/packaging.html#creating-a-pull-request)

### CODE REVIEW

File: ports/fnafnes/FNAF NES.sh [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+elif [ -d "/opt/tools/PortMaster/" ]; then
+  controlfolder="/opt/tools/PortMaster"
+elif [ -d "$XDG_DATA_HOME/PortMaster/" ]; then
+  controlfolder="$XDG_DATA_HOME/PortMaster"
+else
+  controlfolder="/roms/ports/PortMaster"
+fi
+
+source $controlfolder/control.txt
+source $controlfolder/device_info.txt
```
Comment (cdeletre) [inline ports/fnafnes/FNAF NES.sh:14]: The launch script template has been updated see [packaging](https://portmaster.games/packaging.html#the-launchscript-sh) :)
```suggestion
```

File: ports/fnafnes/FNAF NES.sh [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+export PORT_32BIT="Y"
+
+get_controls
+[ -f "${controlfolder}/mod_${CFW_NAME}.txt" ] && source "${controlfolder}/mod_${CFW_NAME}.txt"
```
Comment (cdeletre) [inline ports/fnafnes/FNAF NES.sh:18]: `get_controls` must be called after sourcing the mod file
```suggestion
[ -f "${controlfolder}/mod_${CFW_NAME}.txt" ] && source "${controlfolder}/mod_${CFW_NAME}.txt"
get_controls
```

File: ports/fnafnes/FNAF NES.sh [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+$ESUDO chmod 666 /dev/tty0
```
Comment (cdeletre) [inline ports/fnafnes/FNAF NES.sh:20]: The launch script template has been updated, this is no more needed. See [packaging](https://portmaster.games/packaging.html#the-launchscript-sh) :)

```suggestion
```

File: ports/fnafnes/FNAF NES.sh [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+GAMEDIR="/$directory/ports/fnafnes"
+
+export LD_LIBRARY_PATH="/usr/lib32:$GAMEDIR/libs"
+export GMLOADER_DEPTH_DISABLE=1
+export GMLOADER_SAVEDIR="$GAMEDIR/gamedata/"
+export GMLOADER_PLATFORM="os_linux"
+
+# We log the execution of the script into log.txt
+exec > >(tee "$GAMEDIR/log.txt") 2>&1
```
Comment (cdeletre) [inline ports/fnafnes/FNAF NES.sh:30]: The launch script template has been updated see [packaging](https://portmaster.games/packaging.html#the-launchscript-sh) :)
```suggestion
> "$GAMEDIR/log.txt" && exec > >(tee "$GAMEDIR/log.txt") 2>&1
```

File: ports/fnafnes/FNAF NES.sh [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+cd $GAMEDIR
+
+if [ -f "${controlfolder}/libgl_${CFWNAME}.txt" ]; then 
+  source "${controlfolder}/libgl_${CFW_NAME}.txt"
+else
+  source "${controlfolder}/libgl_default.txt"
+fi
```
Comment (cdeletre) [inline ports/fnafnes/FNAF NES.sh:38]: Is gl4es needed ? Test with libmali on rocknix without these 4 lines to check.

File: ports/fnafnes/FNAF NES.sh [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+# Check for file existence before trying to manipulate them:
+[ -f "./gamedata/data.win" ] && mv gamedata/data.win gamedata/game.droid
+[ -f "./gamedata/game.win" ] && mv gamedata/game.win gamedata/game.droid
+[ -f "./gamedata/game.unx" ] && mv gamedata/game.unx gamedata/game.droid
+
+# Make sure uinput is accessible so we can make use of the gptokeyb controls
+$ESUDO chmod 666 /dev/uinput
```
Comment (cdeletre) [inline ports/fnafnes/FNAF NES.sh:46]: The launch script template has been updated, this is no more needed. See [packaging](https://portmaster.games/packaging.html#the-launchscript-sh) :)
```suggestion
```

File: ports/fnafnes/FNAF NES.sh [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+$GPTOKEYB "gmloader" -c ./fnafnes.gptk &
+
+$ESUDO chmod +x "$GAMEDIR/gmloader"
+
```
Comment (cdeletre) [inline ports/fnafnes/FNAF NES.sh:51]: The launch script template has been updated. See [packaging](https://portmaster.games/packaging.html#the-launchscript-sh) :)

```suggestion
pm_platform_helper "$GAMEDIR/gmloader"
```

File: ports/fnafnes/FNAF NES.sh [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+./gmloader fnafnes.apk
+
+$ESUDO kill -9 $(pidof gptokeyb)
+$ESUDO systemctl restart oga_events &
+printf "\033c" > /dev/tty0
```
Comment (cdeletre) [inline ports/fnafnes/FNAF NES.sh:56]: The launch script template has been updated. Now we just call `pm_finish` See [packaging](https://portmaster.games/packaging.html#the-launchscript-sh) :)
```suggestion
pm_finish
```

File: ports/fnafnes/fnafnes/fnafnes.apk
Comment (cdeletre) [inline ports/fnafnes/fnafnes/fnafnes.apk:1]: good job ! there is no game asset :)

File: ports/fnafnes/cover.png
Comment (cdeletre) [inline ports/fnafnes/cover.png:1]: Please resize to 400 x 300 and optimize file size with `pngquant` or an online image optimizer tool of your choice :)

File: ports/fnafnes/screenshot.png
Comment (cdeletre) [inline ports/fnafnes/screenshot.png:1]: Screenshot must be 640x480

**Review by cdeletre — CHANGES_REQUESTED** [review]
Hallo Damian !

Here are my change requests :) Feel free to DM me for any question.

Kdog

File: ports/fnafnes/FNAF NES.sh [hunk 9]
```diff
@@ -0,0 +1,56 @@
+XDG_DATA_HOME=${XDG_DATA_HOME:-$HOME/.local/share}
```
Comment (cdeletre) [inline ports/fnafnes/FNAF NES.sh:1]:
```suggestion
#!/bin/bash

XDG_DATA_HOME=${XDG_DATA_HOME:-$HOME/.local/share}
```

**Review by cdeletre — CHANGES_REQUESTED** [review]

File: ports/fnafnes/FNAF NES.sh (hunk 6 above)
Comment (cdeletre) [reply ports/fnafnes/FNAF NES.sh:37]: you forgot this one :)

Comment (Damian2809) [reply ports/fnafnes/FNAF NES.sh:37]: fuck

File: ports/fnafnes/fnafnes/libs/libcrypto.so.1.1
Comment (cdeletre) [inline ports/fnafnes/fnafnes/libs/libcrypto.so.1.1:1]: You need to add the licence for this lib. You can reuse this one https://github.com/cdeletre/PortMaster-MiniDoom2/blob/main/minidoom2/LICENSE.libcrypto.txt

File: ports/fnafnes/fnafnes/libs/libzip.so.5
Comment (cdeletre) [inline ports/fnafnes/fnafnes/libs/libzip.so.5:1]: https://github.com/cdeletre/PortMaster-MiniDoom2/blob/main/minidoom2/LICENSE.libzip.txt

Comment (cdeletre) [inline ports/fnafnes/fnafnes/libs/libzip.so.5:1]: You need to add the license for this lib. You car reuse this one https://github.com/cdeletre/PortMaster-MiniDoom2/blob/main/minidoom2/LICENSE.libzip.txt

File: ports/fnafnes/fnafnes/libs/libcrypto.so.1.1
Comment (cdeletre) [inline ports/fnafnes/fnafnes/libs/libcrypto.so.1.1:1]: You need to add the licence for this lib. You can reuse this one https://github.com/cdeletre/PortMaster-MiniDoom2/blob/main/minidoom2/LICENSE.libcrypto.txt

File: ports/fnafnes/port.json [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+  ],
+  "items_opt": [],
+  "attr": {
+    "title": "FNAF NES STYLED",
+    "porter": [
+      "Damian2809"
+    ],
+    "desc": "In FNAF NES STYLED, you play as a night security guard at Freddy Fazbear’s Pizza. Your main goal is to survive five nights while being stalked by animatronic characters that come to life at night. You monitor security cameras to track their movements and use limited resources like door controls and lights to protect yourself from being caught. The game is a mix of strategy and horror, requiring you to manage your resources carefully and stay alert to avoid the animatronics.\n ",
+    "desc_md": null,
+    "inst": "Download the game from https://tai5-5.itch.io/fnafnes then put the data.win into fnafnes/gamedata",
```
Comment (cdeletre) [inline ports/fnafnes/port.json:16]:
```suggestion
    "inst": "Download the game from https://tai5-5.itch.io/fnafnes, unzip FNAFNES.zip then put the data.win into fnafnes/gamedata",
```

**Review by cdeletre — CHANGES_REQUESTED** [review]
Don't forget https://github.com/PortsMaster/PortMaster-New/pull/856/files#r1817450258 :)

**Review by cdeletre — DISMISSED** [review]

File: ports/fnafnes/FNAF NES.sh [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+
+source $controlfolder/control.txt
+export PORT_32BIT="Y"
+
+[ -f "${controlfolder}/mod_${CFW_NAME}.txt" ] && source "${controlfolder}/mod_${CFW_NAME}.txt"
+get_controls
+
+GAMEDIR="/$directory/ports/fnafnes"
+
+export LD_LIBRARY_PATH="/usr/lib32:$GAMEDIR/libs"
```
Comment (Cebion) [inline ports/fnafnes/FNAF NES.sh:21]: Wrong LD_LIBRARY_PATH Export.

**Review by Cebion — CHANGES_REQUESTED** [review]

File: ports/fnafnes/FNAF NES.sh (hunk 11 above)
Comment (Damian2809) [reply ports/fnafnes/FNAF NES.sh]: fixed

Comment (cdeletre) [reply ports/fnafnes/FNAF NES.sh]: you now need to rename `libs` to `libs.armhf` and check that gmloader still works :)

**Review by cdeletre — APPROVED** [review]

**Review by Cebion — APPROVED** [review]

**Pull request merged**